packages/multichain/src/scope/supported.ts

```diff
@@ -0,0 +1,114 @@
+import { toHex, isEqualCaseInsensitive } from '@metamask/controller-utils';
+import type { CaipAccountId, Hex } from '@metamask/utils';
+import { KnownCaipNamespace, parseCaipAccountId } from '@metamask/utils';
+
+import {
+  KnownNotifications,
+  KnownRpcMethods,
+  KnownWalletNamespaceRpcMethods,
+  KnownWalletRpcMethods,
+} from './constants';
+import type { NonWalletKnownCaipNamespace, ExternalScopeString } from './types';
+import { parseScopeString } from './types';
+
+/**
+ * Determines if a scope string is supported.
+ * @param scopeString - The scope string to check.
+ * @param isChainIdSupported - A predicate that determines if a chainID is supported.
+ * @returns A boolean indicating if the scope string is supported.
+ */
+export const isSupportedScopeString = (
+  scopeString: string,
+  isChainIdSupported: (chainId: Hex) => boolean,
+) => {
+  const { namespace, reference } = parseScopeString(scopeString);
+
+  switch (namespace) {
+    case KnownCaipNamespace.Wallet:
+      return !reference || reference === KnownCaipNamespace.Eip155;
+    case KnownCaipNamespace.Eip155:
+      return !reference || isChainIdSupported(toHex(reference));
+    default:
+      return false;
+  }
+};
+
+/**
+ * Determines if an account is supported by the wallet (i.e. on a keyring known to the wallet).
+ * @param account - The CAIP account ID to check.
+ * @param getInternalAccounts - A function that returns the internal accounts.
+ * @returns A boolean indicating if the account is supported by the wallet.
+ */
+export const isSupportedAccount = (
+  account: CaipAccountId,
+  getInternalAccounts: () => { type: string; address: string }[],
+) => {
+  const {
+    address,
+    chain: { namespace, reference },
+  } = parseCaipAccountId(account);
+
+  const isSupportedEip155Account = () =>
+    getInternalAccounts().some(
+      (internalAccount) =>
+        ['eip155:eoa', 'eip155:erc4337'].includes(internalAccount.type) &&
+        isEqualCaseInsensitive(address, internalAccount.address),
+    );
+
+  switch (namespace) {
+    case KnownCaipNamespace.Wallet:
+      return reference === KnownCaipNamespace.Eip155
+        ? isSupportedEip155Account()
```

Comment:
This branch seems to be meant to support `wallet:eip155:[account]` addresses, which the tests confirm. This type of account specifier seems..... questionable, because it doesn't specify which chain the account exists on. For a non-EOA EVM account, it might only exist on one chain, but this identifier wouldn't have that information.

It's my understanding that this type of account specifier is intended as a temporary solution for supporting snaps? It's unclear exactly what this is needed for, but it might be worth creating an issue or comment or both to follow-up on this later. We should at the very least ensure this type of account specifier is well documented, and that the risks are known. Or we could find a way to get rid of it.

Reply:
>I'm not entirely sure if this matters since this doesn't actually grant any permissions to do any writes against any chain (on which the SCA was not deployed say) since those permissions would have to be granted explicitly within a chain defined scope.

Right.... good point 🤔 OK, still seems less than ideal to have two representations, one with less information, but you're right. No obvious reason for concern. Resolving.